### PR TITLE
updated error calls with module names

### PR DESCRIPTION
**Pull request by alizahidraja**
## Description

Included Module names in channel callback errors

closes: #1728

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing) (does not apply)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`) (does not apply)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by alizahidraja** [comment]
Will update the code to fix for issue raised by @colin-axner 

> Thanks @alizahidraja! Unfortunately with middleware, these error messages can be misleading. If we have a middleware stack which is ICS29 and ICS20, if ICS20 returns an error in the application callback, the final error message will actually indicate that the module which failed was fee-middleware (which is incorrect).
> 
> I think it might make more sense to include the portID? We could also change the notion of the "module" variable to be a "middleware stack", though it'd be nice to think this approach out fully

**Comment by alizahidraja** [comment]
I have mentioned both the portID & channelID in channel callback errors for better identification, lmk if that works

File: modules/core/keeper/msg_server.go [hunk 1]
```diff
@@ -190,7 +190,7 @@ func (k Keeper) ChannelOpenInit(goCtx context.Context, msg *channeltypes.MsgChan
 	// Perform application logic callback
 	version, err := cbs.OnChanOpenInit(ctx, msg.Channel.Ordering, msg.Channel.ConnectionHops, msg.PortId, channelID, cap, msg.Channel.Counterparty, msg.Channel.Version)
 	if err != nil {
-		return nil, sdkerrors.Wrap(err, "channel open init callback failed")
+		return nil, sdkerrors.Wrapf(err, "channel open init callback failed for Port ID: %s, Channel ID: %s", msg.PortId, channelID)
```
Comment (colin-axner) [inline modules/core/keeper/msg_server.go:193]:
```suggestion
		return nil, sdkerrors.Wrapf(err, "channel open init callback failed for port ID: %s, channel ID: %s", msg.PortId, channelID)
```

I think most of our errors avoid capitals so I think it'd make sense to be consistent

**Review by colin-axner — APPROVED** [review]
Thanks @alizahidraja for following up on my suggestion. I think this should help make the error message a bit more useful. Just left a minor suggestion on the error string formatting, but I don't have a strong preference here

File: modules/core/keeper/msg_server.go (hunk 1 above)
Comment (alizahidraja) [reply modules/core/keeper/msg_server.go:193]: That's great! will convert these to lower case

File: CHANGELOG.md [hunk 2]
```diff
@@ -102,6 +102,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (apps/27-interchain-accounts) [\#2248](https://github.com/cosmos/ibc-go/pull/2248) Adding call to underlying app in `OnChanCloseConfirm` callback of the controller submodule and adding relevant unit tests.
 * (apps/27-interchain-accounts) [\#2251](https://github.com/cosmos/ibc-go/pull/2251) Adding `msgServer` struct to controller submodule that embeds the `Keeper` struct.    
 * (apps/27-interchain-accounts) [\#2297](https://github.com/cosmos/ibc-go/pull/2297) Adding cli command to generate ICS27 packet data.
+* (modules/core/keeper) [\#1728](https://github.com/cosmos/ibc-go/pull/2399) Updated channel callback errors to includ portID & channelID for better identification of errors.
```
Comment (damiannolan) [inline CHANGELOG.md:105]: typo! 

```suggestion
* (modules/core/keeper) [\#1728](https://github.com/cosmos/ibc-go/pull/2399) Updated channel callback errors to include portID & channelID for better identification of errors.
```

**Review by damiannolan — APPROVED** [review]
Thanks @alizahidraja! LGTM 🚀

**Review by chatton — APPROVED** [review]
LGTM!

**Pull request merged**